#HYPOTHESIS TESTING

In [4]:
#LIBRARIES NEEDED
import numpy as np
import pandas as pd
from scipy import stats as sts
import warnings
warnings.filterwarnings("ignore") #just to ignore any warnings

In hypothesis tetsing we have two kinds of hypothesis statements
-  Null Hypothesis(H0) : when there is no significant difference between the old process and the newly implemented method
                    μA=μB  OR μA>=μB
- Alternative Hypothesis(Ha) : when there is a significant difference between the old process and newly implemented process
                    μA!=μB  OR μA<μB

#FOR THE GIVEN CASE STUDY, LETS ASSUME THE HYPOTHESIS STATEMENTS

#1)State the Hypotheses statement
Null Hypothesis(H0) : The mean weekly operating cost is equal to the theoretical mean weekly cost according to the cost model.
- H0: μ = 1000+5*600 = 4000

Alternative Hypothesis(Ha) :The mean weekly operating cost is higher than the theoretical mean weekly cost according to the cost model.
- Ha: μ > 4000

#2)Calculate the Test Statistic:
Given
- sample mean (x) = 3050
- theoritical mean(μ) = 4000
- standard deviation = 5*25=125
- sample size (n) = 25

In [2]:
# calculating the t test values as the smaple size is big enough we can use any test method, here we are using t test
#lets define the values and use them in the know formula of t test

x=3050
mu=4000
sigma=125
n=25

tts=(x-mu)/((sigma/n**0.5))
print(tts)

-38.0


#3)Determine the Critical Value:
For a one-tailed test at the alpha level of 5% (α = 0.05), we refer to the standard normal (Z) distribution table to find the critical value.

Since it's a one-tailed test with α = 0.05:

using this value of alpha level , lets calculate the probability using stats module

In [5]:
p=sts.t.cdf(tts,df=24)
alpha=0.05
p

2.9607810808177907e-23

#4)Make a Decision:

The cumulative probability for a t-score of -38 with 24 degrees of freedom is approximately 2.96*10^-23. This probability is extremely small, indicating that the likelihood of observing a value as extreme as -38 in a t-distribution with 24 degrees of freedom is virtually zero. This confirms that the test statistic is far beyond the critical value, providing strong evidence against the null hypothesis. ​

In [10]:
if p>alpha:
  print("accept the null hypothesis and do not launch the program")
else:
  print("fail to accept null hypotesis and launch the program")

# we are using the probability error and comparing it with the level of significance alpha and making a decision to collide with the theoritical explanation

fail to accept null hypotesis and launch the program


#5) Conclusion:
Based on the decision in step 4, we conclude that there is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests. The observed weekly operating costs are significantly different from the theoretical costs predicted by the model, indicating that the actual costs are higher than the model estimates.

#CHI SQUARE TESTING

In [13]:
#LIBRARIES NEEDED ARE ALREADY IMPORTED, JUST IN CASE

import numpy as np
import pandas as pd
from scipy import stats as sts
from scipy.stats import chi2_contingency
import warnings
warnings.filterwarnings("ignore") #just to ignore any warnings

In CHI-SQUARE hypothesis testing we have two kinds of tests
- **Chi-square test of independence**: Used to determine if there is a significant association between two categorical variables.
- **Chi-square goodness-of-fit test**: Used to determine if a sample data matches a population with a specific distribution.

As per the question, conclude that we are here going to perform the **Chi-square test of independence**



 hypothesis statements
-  Null Hypothesis(H0) : this hypothesis tells that the variables are independent and have no significant association to each other
- Alternative Hypothesis(Ha) : this hypothesis tells that the variables are dependent and have a significant association to each other

#FOR THE GIVEN CASE STUDY, LETS ASSUME THE HYPOTHESIS STATEMENTS

Hypothesis Statements

* Null Hypothesis(H0) : there is no significant association between the type of device purchased and the customer's satisfaction level.

* Alternative Hypothesis(Ha) : there's a significant association between the type of device purchased and the customer's satisfaction level.

In [12]:
data=np.array([[50,70],[80,100],[60,90],[30,50],[20,50]])
print(data)

[[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]


#COMPUTE THE CHI-SQUARE STATISTIC VALUE

In [26]:
chisq=chi2_contingency(data)
print(chisq,"\n")
print("the expected table is :", "\n","\n",chisq[3])
# using chi2_contingency we found the required information, the calculated chi square statistic value,pvalue, degrees of freedom, expected values

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]])) 

the expected table is : 
 
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [23]:
print("The Calculated chi2_stat:",chisq[0])
print("The Probability is:",chisq[1])
print("The Degrees of Freedom:",chisq[2])

The Calculated chi2_stat: 5.638227513227513
The Probability is: 0.22784371130697179
The Degrees of Freedom: 4


In [27]:
#LETS AVE PROBABILITY TO A VARIABLE SO WE CAN USE IT TO ANY OTHER COMPARISIONS
p=chisq[1]

#DETERMINE THE CRITICAL VALUE

the level of significance is alpha=0.05

#MAKING A DECISION

In [28]:
if p <= alpha:
    print("Accept the Null Hypothesis.")
else:
    print("Reject the Null Hypothesis.")

Reject the Null Hypothesis.


#CONCLUSIONS

In [30]:
if p <= alpha:
    print("There is no significant association between the type of device purchased and the customer's satisfaction level.")
else:
    print("There is a significant association between the type of device purchased and the customer's satisfaction level.")

There is a significant association between the type of device purchased and the customer's satisfaction level.
